File: backend/app/schemas/contract.py
import uuid
from datetime import date, datetime
from typing import Optional, List
from pydantic import BaseModel
from app.models.contract import ContractStatus


class ExtractedFieldsOut(BaseModel):
    parties: Optional[List[str]] = None
    effective_date: Optional[date] = None
    expiry_date: Optional[date] = None
    value: Optional[float] = None
    currency: Optional[str] = None
    governing_law: Optional[str] = None
    renewal_terms: Optional[str] = None
    key_clauses: Optional[List[str]] = None

    class Config:
        from_attributes = True


class ContractOut(BaseModel):
    id: uuid.UUID
    file_name: str
    status: ContractStatus
    uploaded_at: datetime
    extracted_fields: Optional[ExtractedFieldsOut] = None
    raw_text: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/contract.ts
export interface ExtractedFields {
  parties: string[] | null;
  effective_date: string | null;
  expiry_date: string | null;
  value: number | null;
  currency: string | null;
  governing_law: string | null;
  renewal_terms: string | null;
  key_clauses: string[] | null;
}

export type ContractStatus = "uploaded" | "processing" | "extracted" | "failed";

export type LifecycleStatus =
  | "draft"
  | "active"
  | "executed"
  | "expired"
  | "terminated";

export interface Contract {
  id: string;
  file_name: string;
  status: ContractStatus;
  uploaded_at: string;
  extracted_fields: ExtractedFields | null;
  is_starred: boolean;
  lifecycle_status: LifecycleStatus;
}